openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -298,24 +301,31 @@ def _track_update_email_opt_in(user_id, organization, opt_in):
     )
 
 
-def _get_user(requesting_user, username=None, allow_staff=False):
+def _get_authorized_user(requesting_user, username=None, allow_staff=False):
     """
-    Helper method to return the user for a given username.
+    Helper method to return the authorized user for a given username.
     If username is not provided, requesting_user.username is assumed.
     """
     if username is None:
         username = requesting_user.username
-
     try:
         existing_user = User.objects.get(username=username)
     except ObjectDoesNotExist:
         raise UserNotFound()
 
+    _is_authorized(requesting_user, username, allow_staff)
+    return existing_user
+
+
+def _is_authorized(requesting_user, username, allow_staff=False):
```

Comment:
Nit: since the return value of this method is never used, I would instead rename it as check_authorized or verify_authorized and not bother returning anything (as it is, it's unclear if the return value needs to be checked).